### ISSUE TITLE
Allow --test against a glob pattern of directories

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#2064]
Instead of `yarn test:rules` to run them all, or manually modifying `ruleTestRunner.js` to change what's running, it'd be nice to quickly run `yarn test:rules -- no-floating-promises/*`.

It could be done pretty easily in `ruleTestRunner.ts` with:

```typescript
// Abridged for ease of reading
const paths = process.argv[2] || "**";
glob.sync(`test/rules/${paths}/tslint.json`)
```

**Comment by mohsen1** [comment]
You can use tslint CLI to test a specific rule. See `--test` flag in help. If you are making changes to tslint itself, you can compile and use the new compiled CLI

**Comment by JoshuaKGoldberg** [comment]
> making changes to tslint

@mohsen1 yes, that's what I'm doing now. I'd like to sporadically run `test:rules` on *specific* files without having to recompile the CLI.

**Comment by JoshuaKGoldberg** [comment]
To be a little more elaborate - `--test` only runs against one directory. 

```cmd
C:\Code\tslint>tslint --test test/rules/no-floating-promises/**
C:\Code\tslint\lib\configuration.js:114
            throw new Error("Could not find config file at: " + path.resolve(suppliedConfigFilePath));
            ^

Error: Could not find config file at: C:\Code\tslint\test\rules\no-floating-promises\**\tslint.json
```

It would be nice to have a little more control and be able to test multiple directories. Perhaps letting `--test` run against multiple directories would accomplish the same thing...

**Comment by nchen63** [comment]
I have a slight preference for `--test` that works against more than 1 directory

**Comment by JoshuaKGoldberg** [comment]
That makes sense, they're basically the same. Updated the title.
